api/samples/memtrace_simple.c

```diff
@@ -418,6 +435,19 @@ dr_client_main(client_id_t id, int argc, const char *argv[])
     drreg_options_t ops = { sizeof(ops), 3, false };
     dr_set_client_name("DynamoRIO Sample Client 'memtrace'",
                        "http://dynamorio.org/issues");
+
+    const char *client_ops = dr_get_options(id);
+    if (client_ops != NULL && client_ops[0] != '\0') {
+        if (strcmp(client_ops, "-log_to_stderr") == 0)
+            log_to_stderr = true;
+        else {
+            dr_fprintf(STDERR,
```

Comment:
The `!log_to_stderr` case (dump data to a per-thread file) seems to be supported above though.

Reply:
You mean the message "unknown options |%s|: only -log_to_stderr is supported" implies that passing no options at all is *not* supported?  The message will not be printed if there are no options though.  Is this what you mean?  That it should say "only -log_to_stderr or an empty string are supported"?

Reply:
> The message will not be printed if there are no options though

Right, I see that.

I couldn't understand why we want to fail in this specific scenario (if options are given but not including `-log_to_stderr`). Do we want the user to not pass any other option? Why? One could pass other options AND `-log_to_stderr`, and that'd be just as wrong, but the current implementation doesn't fail in that case.

What if we add some new option in future and the user wants to dump data to a per-thread file?

Reply:
> One could pass other options AND -log_to_stderr, and that'd be just as wrong, but the current implementation doesn't fail in that case.

I guess failing if the user passes unsupported options is reasonable. But let's handle the above case too.

Reply:
> One could pass other options AND -log_to_stderr, and that'd be just as wrong, but the current implementation doesn't fail in that case.

It would fail in that case.  It's doing a strcmp on the entire option string, not just one token: there is no tokenization.

Reply:
I just noticed that dr_get_options() is deprecated.  So maybe I shouldn't use it in a provided sample.  Supposed to use the argv[] passed in -- forgot about that.  Will rewrite using argv[] and again only allowing no options or just this one.

Reply:
> It would fail in that case. It's doing a strcmp on the entire option string, not just one token: there is no tokenization.

Oops. sorry. Makes sense now.